ip_list.go

```diff
@@ -0,0 +1,445 @@
+	Kind                  string     `json:"kind"`
+	NumItems              int        `json:"num_items"`
+	NumReferencingFilters int        `json:"num_referencing_filters"`
+	CreatedOn             *time.Time `json:"created_on"`
+	ModifiedOn            *time.Time `json:"modified_on"`
+}
+
+// IPListItem contains information about a single IP List Item
+type IPListItem struct {
+	ID         string     `json:"id"`
+	IP         string     `json:"ip"`
+	Comment    string     `json:"comment"`
+	CreatedOn  *time.Time `json:"created_on"`
+	ModifiedOn *time.Time `json:"modified_on"`
+}
+
+// IPListCreateRequest contains data for a new IP List
+type IPListCreateRequest struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Kind        string `json:"kind"`
+}
+
+// IPListItemCreateRequest contains data for a new IP List Item
+type IPListItemCreateRequest struct {
+	IP      string `json:"ip"`
+	Comment string `json:"comment"`
+}
+
+// IPListItemDeleteRequest wraps IP List Items that shall be deleted
+type IPListItemDeleteRequest struct {
+	Items []IPListItemDeleteItemRequest `json:"items"`
+}
+
+// IPListItemDeleteItemRequest contains single IP List Items that shall be deleted
+type IPListItemDeleteItemRequest struct {
+	ID string `json:"id"`
+}
+
+// IPListUpdateRequest contains data for an IP List update
+type IPListUpdateRequest struct {
+	Description string `json:"description"`
+}
+
+// IPListResponse contains a single IP List
+type IPListResponse struct {
+	Response
+	Result IPList `json:"result"`
+}
+
+// IPListItemCreateResponse contains information about the creation of an IP List Item
+type IPListItemCreateResponse struct {
+	Response
+	Result struct {
+		OperationID string `json:"operation_id"`
+	} `json:"result"`
+}
+
+// IPListListResponse contains a slice of IP Lists
+type IPListListResponse struct {
+	Response
+	Result []IPList `json:"result"`
+}
+
+// IPListBulkOperationResponse contains information about a Bulk Operation
+type IPListBulkOperationResponse struct {
+	Response
+	Result IPListBulkOperation `json:"result"`
+}
+
+// IPListDeleteResponse contains information about the deletion of an IP List
+type IPListDeleteResponse struct {
+	Response
+	Result struct {
+		ID string `json:"id"`
+	} `json:"result"`
+}
+
+// IPListItemsListResponse contains information about IP List Items
+type IPListItemsListResponse struct {
+	Response
+	ResultInfo `json:"result_info"`
+	Result     []IPListItem `json:"result"`
+}
+
+// IPListItemDeleteResponse contains information about the deletion of an IP List Item
+type IPListItemDeleteResponse struct {
+	Response
+	Result struct {
+		OperationID string `json:"operation_id"`
+	} `json:"result"`
+}
+
+// IPListItemsGetResponse contains information about a single IP List Item
+type IPListItemsGetResponse struct {
+	Response
+	Result IPListItem `json:"result"`
+}
+
+// ListIPLists lists all IP Lists
+// API reference: https://api.cloudflare.com/#rules-lists-list-lists
```

Comment:
In the interest of making the code comments for methods all consistent, can we add a newline between the description and API reference lines?

```suggestion
// ListIPLists lists all IP Lists
//
// API reference: https://api.cloudflare.com/#rules-lists-list-lists
```